src/sqlBible/src/SqlBible.ts

```diff
@@ -0,0 +1,292 @@
+        return this.getNextPhraseIdForNormalizedVerseNum(nRef, reference.versionId);
+    }
+
+    getNormalizedReference(reference: IBibleReferenceRangeVersion): IBibleReferenceRangeNormalized {
+        // setting all missing properties on reference
+        if (!reference.versionChapterEndNum)
+            reference.versionChapterEndNum =
+                reference.versionChapterNum || BOOK_DATA[reference.bookOsisId].chapters.length;
+        if (!reference.versionChapterNum) reference.versionChapterNum = 1;
+        if (!reference.versionVerseNum) reference.versionVerseNum = 1;
+        if (!reference.versionVerseEndNum)
+            reference.versionVerseEndNum =
+                BOOK_DATA[reference.bookOsisId].chapters[reference.versionChapterEndNum - 1];
+
+        // casting to required type since we are sure that every property is set now
+        type reqRef = Required<IBibleReferenceRangeVersion>;
+        const { normalizedChapterNum, normalizedVerseNum } = this.getNormalizedV11n(<reqRef>(
+            reference
+        ));
+        const {
+            normalizedChapterNum: normalizedChapterEndNum,
+            normalizedVerseNum: normalizedVerseEndNum
+        } = this.getNormalizedV11n({
+            versionId: reference.versionId,
+            bookOsisId: reference.bookOsisId,
+            versionChapterNum: reference.versionChapterEndNum,
+            versionVerseNum: reference.versionVerseEndNum
+        });
+
+        return {
+            bookOsisId: reference.bookOsisId,
+            normalizedChapterNum,
+            normalizedVerseNum,
+            normalizedChapterEndNum,
+            normalizedVerseEndNum
+        };
+    }
+
+    getNormalizedV11n(
+        reference: Required<IBibleReferenceVersion>
+    ): Required<IBibleReferenceNormalized> {
+        // TODO: normalize this using the v11n-normalisation data from STEPData
+        return {
+            bookOsisId: reference.bookOsisId,
+            normalizedChapterNum: reference.versionChapterNum,
+            normalizedVerseNum: reference.versionVerseNum
+        };
+    }
+
+    async getPhrases(reference: IBibleReferenceRangeVersion) {
+        await this.dbReady;
+        const nRef = this.getNormalizedReference(reference);
+        return getManager().find(BiblePhrase, {
+            where: {
+                id: Raw(
+                    col =>
+                        `${col} BETWEEN '${generateNormalizedRefId(
+                            nRef,
+                            5
+                        )}' AND '${generateNormalizedRefId(
+                            {
+                                bookOsisId: nRef.bookOsisId,
+                                normalizedChapterNum: nRef.normalizedChapterEndNum,
+                                normalizedVerseNum: nRef.normalizedVerseEndNum
+                            },
+                            5
+                        )}' AND cast(${col} % 100000000000 / 100000000 as int) = ${
```

Comment:
for reasons of overall simplicity and efficiency we encode a reference in a single integer in the database. parsing and encoding this integer might not look nice but it shouldn't make any dent on the CPU (on the contrary it will perform better than juggling around multiple a set of numbers for each reference on different levels of the software). 

i included more helper functions and did some refactoring to hide the modular math away and/or make more explicit what is happening.